### PR TITLE
autocomplete: Use user shape in autocomplete

### PR DESCRIPTION
**Pull request by patochectp**

### CODE REVIEW

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py [hunk 1]
```diff
@@ -57,14 +57,18 @@ def __init__(self, **kwargs):
...
-    def get(self, request, instance):
+    def get(self, request, instance, shape):
         if not self.external_api:
             raise TechnicalError('global autocomplete not configured')
 
         q = request['q']
         url = '{endpoint}?q={q}'.format(endpoint=self.external_api, q=q)
         try:
-            raw_response = requests.get(url, timeout=self.timeout)
+            if shape:
```
Comment (pbougue) [inline source/jormungandr/jormungandr/autocomplete/geocodejson.py:67]: Not sure: shouldn't you test `shape is not None` (and default shape to None)
@kinnou02 what's the pythonic way?

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 2]
```diff
@@ -247,7 +247,8 @@ def get(self, region=None, lon=None, lat=None):
             response = i_manager.dispatch(args, "places", instance_name=instance)
         else:
             if global_autocomplete:
-                bragi_response = global_autocomplete.get(args, None)
+                user = authentication.get_user(token=authentication.get_token())
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: I'd say add a try around that, and if no user or problem, then shape=None

File: source/jormungandr/jormungandr/interfaces/v1/Places.py [hunk 3]
```diff
@@ -247,7 +247,8 @@ def get(self, region=None, lon=None, lat=None):
...
+                bragi_response = global_autocomplete.get(args, None, user.shape)
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/Places.py]: would be nice to name params (`shape=user.shape`)

File: source/jormungandr/jormungandr/interfaces/v1/test/elastic_search_tests.py [hunk 4]
```diff
@@ -452,7 +459,16 @@ def bragi_call_test():
...
+        raw_response = bragi.get({'q': 'rue bobette'}, instance=None, shape=None)
+        places = get_response(raw_response).get('places')
+        assert len(places) == 4
+        bragi_house_jaures_response_check(places[0])
+        bragi_house_lefebvre_response_check(places[1])
+        bragi_street_response_check(places[2])
+        bragi_admin_response_check(places[3])
+
+    with mock.patch('requests.post', mock_requests.get):
+        raw_response = bragi.get({'q': 'rue bobette'}, instance=None, shape=geojson())
```
Comment (pbougue) [inline source/jormungandr/jormungandr/interfaces/v1/test/elastic_search_tests.py:471]: would be nice to be able to check geojson (in mock, or send it back and check outside)

File: source/jormungandr/jormungandr/autocomplete/geocodejson.py (hunk 1 above)
Comment (kinnou02) [reply source/jormungandr/jormungandr/autocomplete/geocodejson.py:67]: `if shape:` is fine

File: source/jormungandr/jormungandr/interfaces/v1/Places.py (hunk 2 above)
Comment (pbougue) [reply source/jormungandr/jormungandr/interfaces/v1/Places.py]: get_user may raise or return None. So try around that, but you'll have to check that it's not None also.

File: source/jormungandr/jormungandr/interfaces/v1/test/elastic_search_tests.py (hunk 4 above)
Comment (pbougue) [reply source/jormungandr/jormungandr/interfaces/v1/test/elastic_search_tests.py:471]: won't be done there

**Review by pbougue — APPROVED** [review]

**Comment by pbougue** [comment]
Jenkins, test this please

**Comment by patochectp** [comment]
Jenkins, test this please

**Comment by pbougue** [comment]
Jenkins, test this please

**Pull request merged**